### PR TITLE
#147322097 Enable Gym Members Comparison

### PR DESCRIPTION
**Pull request by dmkitui**
#### What does this PR do?
Enable the comparison of any two members of the same gym while training together. This feature is available to and is used by the gym trainer, or administrator to have a visual indicator how any two members compare with each other in terms of height, weight, age, work/free hours, calory intakes etc. 
#### Description of Task to be completed?

1. Add access via a button on the gym overview to the user comparison page.
2. Ability to select any active user so that their data can be available on the graph. One can select two users at any one time, selecting a third deselects the earlier most selected member, and allows comparison of the last two.
3. Show the selected users' data side by side in the form of animated bar graphs.

#### How should this be manually tested?

1. Clone the repository.
2 . cd into the wger_argadianz directory and start the server via:
`python manage.py runserver`
3. Ensure presence of at least three gym members with their data updated.
4. On the gym overview page, click on the *Compare Members* button.
5. Click on any two members.
6. Their data should now be displayed on a graph to the left.
The graphs are colour-coded, and the key clearly indicates to whom the metrics belong to.
7. Selecting a third user should deselect one of the users, and pull up the now selected users's data.

#### Any background context you want to provide?
Currently there is no way to display this information.
#### What are the relevant pivotal tracker stories?
#147322097
#### Screenshots (if appropriate)
screen shot 2017-07-14 at 15 13 20
screen shot 2017-07-14 at 15 13 06
screen shot 2017-07-14 at 15 13 29
screen shot 2017-07-14 at 15 13 35
screen shot 2017-07-14 at 15 13 38
screen shot 2017-07-14 at 15 13 42
screen shot 2017-07-14 at 15 13 45
screen shot 2017-07-14 at 15 13 48
screen shot 2017-07-14 at 15 13 52
screen shot 2017-07-14 at 15 13 56

### CODE REVIEW

File: wger/gym/views/gym.py [hunk 1]
```diff
@@ -172,6 +174,57 @@ def get_context_data(self, **kwargs):
...
+        for member in UserProfile.objects.all():
+            members['members'].append(member)
+
+        for u in Gym.objects.get_members(self.kwargs['pk']).select_related('usercache'):
+            members['users'].append({'obj': u})
+        return members
+
+    def get_context_data(self, **kwargs):
+        """Pass other info to the template."""
+        # out2 = UserProfile.objects.filter(user_id=10).all()
```
Comment (andela-sjames) [inline wger/gym/views/gym.py]: Remove commented code from script

File: wger/gym/views/gym.py [hunk 2]
```diff
@@ -56,6 +57,7 @@
     WgerDeleteMixin,
     WgerMultiplePermissionRequiredMixin)
 from wger.utils.helpers import password_generator
+import json
```
Comment (andela-sjames) [inline wger/gym/views/gym.py]: `import json`  should be moved to the top of the script ```Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.``` [reference](https://www.python.org/dev/peps/pep-0008/#imports)

File: wger/gym/views/gym.py [hunk 3]
```diff
@@ -172,6 +174,57 @@ def get_context_data(self, **kwargs):
...
+        # out2 = UserProfile.objects.filter(user_id=10).all()
+        context = super(GymMemberComparisonView, self).get_context_data(**kwargs)
+        context['gym'] = Gym.objects.get(pk=self.kwargs['pk'])
+        context['name'] = "Members Comparison"
+        context['user_table'] = {'keys': [_('ID'), _('Username'), _('Select')]}
+        context['members'] = {'members': context['object_list']['members'],
+                              'users': context['object_list']['users']}
+        data = {}
+        for member in context['members']['members']:
+            if member.calories:
```
Comment (andela-sjames) [inline wger/gym/views/gym.py]: This can be wrapped into a ternary operation [reference](http://book.pythontips.com/en/latest/ternary_operators.html)

**Comment by andela-sjames** [comment]
@dmkitui please review changes suggested on you PR.

**Comment by dmkitui** [comment]
Will do so promptly. Thank you.

Regards,

Daniel Masake Kitui / Andela.com
Software Developer / 254.721.634.281

File: wger/gym/views/gym.py (hunk 2 above)
Comment (dmkitui) [reply wger/gym/views/gym.py]: Done, and reference link appreciated.

File: wger/gym/views/gym.py (hunk 1 above)
Comment (dmkitui) [reply wger/gym/views/gym.py]: Done.

**Pull request merged**